### DISCUSSION TITLE
What does IsarLinks deprecation look like?

### DISCUSSION THREAD

**Discussion by basselalsayed** [isar/isar#1422] [Help]
Hey, 

First of all I'd like to say thanks for maintaining this awesome library, it's a very cool project. 

I've put some initial isar usage into our app, waiting for v4 to implement fully allowing for a complete offline experience. 

I'm curious about what the switch from Links to embedded objects looks like? 

I really like isar's interface and simplicity so would prefer to keep this database instead of switching to something relational, however, I do have some small relation requirements. 

All I need is for two instances of classA to have a relation to one instance of classB, will this still be possible? I see no easy way of doing this with v3 embedded objects.

**Comment by magicmatt007** [comment]
Could we also get some background information, why IsarLinks is not available anymore in v4?

**Comment by shrijanRegmi** [comment]
Some people are waiting for the answer to this question for months..

**Comment by yakou32** [comment]
Same here: I didn't use embedded objects, because with deep nesting, I would have to filter through thousands of items, while querying directly the right level/link gets only a bunch of relevant items.
When syncing with remote sources, it's not always useful to sync the full tree of data with all embedded branches, while syncing per links/ relevant branches is cheaper.

Maybe am I not aware of better practice to use embedded objects for these use cases though...

**Comment by yakou32** [comment]
Hello again :)
I'm trying to mimic a many-to-one relationship.
For instance, a Stage (embedded deeper in the tree) can have only one Photo (collection). And a Photo can belong to several stages.

Photo is a collection because I want to be able to check which stages it belongs to.
It should also be embedded in a stage.

Without using Links, I'm wondering how to solve that issue. One way would be to create an intermediate embedded object "PhotoLinks", that would map the relationships:
- photoLinkId
- photoId
- stageId
It's the (old) way to manage many-to-many relationships.

I'm afraid this is too complex. - code is less readable as it will look like this:
- What is the photo of a stage? Stage -> PhotoLinks where stageId (supposingly one result only) ->  Photo
- In which stage(s) is a photo used? -> PhotoLinks where photoId -> Stages

Would you have any other (hopefully better) ideas?
